src/hugr/rewrite/consts.rs

```diff
@@ -0,0 +1,231 @@
+//! Rewrite operations involving Const and LoadConst operations
+
+use std::iter;
+
+use crate::{
+    hugr::{HugrError, HugrMut},
+    HugrView, Node,
+};
+#[rustversion::since(1.75)] // uses impl in return position
+use itertools::Itertools;
+use thiserror::Error;
+
+use super::Rewrite;
+
+/// Remove a [`crate::ops::LoadConstant`] node with no outputs.
+#[derive(Debug, Clone)]
+pub struct RemoveConstIgnore(pub Node);
+
+/// Error from an [`RemoveConstIgnore`] operation.
+#[derive(Debug, Clone, Error, PartialEq, Eq)]
+pub enum RemoveConstIgnoreError {
+    /// Invalid node.
+    #[error("Node is invalid (either not in HUGR or not LoadConst).")]
+    InvalidNode(Node),
+    /// Node in use.
+    #[error("Node: {0:?} has non-zero outgoing connections.")]
+    ValueUsed(Node),
+    /// Not connected to a Const.
+    #[error("Node: {0:?} is not connected to a Const node.")]
```

Comment:
nit: this does raise the question of, if this has happened, what the heck are you supposed to do with it ;-). I guess you have to remind to a state where you can avoid getting into that mess. Consider defining `ApplyResult = Option<Node>` and then `return None` in such cases?

Reply:
Don't we necessarily fail validation if this is the case? It is too much to ask every rewrite to handle and report every error that would be detected by validate().

Reply:
yes I removed this error case